src/StreamJsonRpc.Tests/MessagePackFormatter.cs

```diff
@@ -119,5 +171,95 @@ public void Serialize(IBufferWriter<byte> contentBuffer, JsonRpcMessage message)
 
             return result;
         }
+
+        private class StandardPlusIProgressOfTResolver : IFormatterResolver
+        {
+            private readonly MessagePackFormatter formatter;
+
+            public StandardPlusIProgressOfTResolver(MessagePackFormatter formatter)
+            {
+                this.formatter = formatter;
+            }
+
+            public IMessagePackFormatter<T> GetFormatter<T>()
+            {
+                return new FormatterCache<T>(this.formatter).Formatter;
+            }
+
+            private class FormatterCache<T>
+            {
+                public readonly IMessagePackFormatter<T> Formatter;
+
+                public FormatterCache(MessagePackFormatter formatter)
+                {
+                    if (MessageFormatterHelper.FindIProgressOfT(typeof(T)) != null)
+                    {
+                        // Call Get Formatter from IProgress Formatter?
+                        this.Formatter = new IProgressOfTFormatter<T>(formatter);
+                    }
+                    else
+                    {
+                        this.Formatter = StandardResolver.Instance.GetFormatter<T>();
+                    }
+                }
+            }
+        }
+
```

Comment:
You don't have to write your own resolver. You can use the CompositeResolver that's built into the MessagePack library.